js/services/Rules.ts

```diff
@@ -69,13 +69,16 @@ const Rules = {
         let result = false;
 
         for (const rule of rules) {
-            const contentType = manywho.component.contentTypes.string;
 
             let left = snapshot.getValue(rule.leftValueElementToReferenceId);
-            left = getStateValue(rule.leftValueElementToReferenceId, left.typeElementId, left.contentType, '') || left;
+            // TODO - What to do if the operands have different contentTypes ? E.g.
```

Comment:
I'm not 100% sure on this one. I think when trying to compare objects/lists with primitives, then an exception should probably be thrown. I'm not sure whether the expected behaviour is to use type coercion on comparing primitives or whether we support a deep diff on comparing objects.

Reply:
Going to leave this one for now and tackle it later if it becomes an issue.  There's a lot of refactoring and error checking work to do, let alone work out the semantics of type conversions.